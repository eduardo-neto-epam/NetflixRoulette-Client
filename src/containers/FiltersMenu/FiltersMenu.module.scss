.menu {
    display: inline-flex;
    flex-direction: row;
    padding: 0;
    margin: 0;
}

.filter {
    list-style: none;
}

.filter + .filter {
    padding-left: 20px;
}