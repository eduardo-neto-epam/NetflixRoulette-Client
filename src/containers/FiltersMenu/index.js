import React, { useState, useEffect } from 'react'
import styles from './FiltersMenu.module.scss'
import classnames from 'classnames/bind'

import FilterSelector from '../../components/FilterSelector'

let cx = classnames.bind(styles)

const FiltersMenu = () => {

    const filters = [
        'all',
        'documentary',
        'comedy',
        'horror',
        'crime',
    ]

    const [filter, setFilter] = useState('all')

    function handleSelect(e) {
        setFilter(e.target.value)
    }

    useEffect(()=>{
        console.log('Current Filter is: ', filter)
    }, [filter])

    return (
        <>
        <ul className={cx('menu')}>
            { filters.map((filter, index) => (
                <li 
                    key={index} 
                    className={cx(
                        'filter'
                        )}>
                    <FilterSelector
                     handleSelect={handleSelect}
                     >
                         {filter}
                    </FilterSelector>
                </li>
            )) }
        </ul>
        </>
    )
}

export default FiltersMenu