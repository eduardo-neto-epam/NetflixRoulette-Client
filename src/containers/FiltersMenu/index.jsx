import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames/bind';
import { v4 as uuidv4 } from 'uuid';
import styles from './FiltersMenu.module.scss';

import FilterSelector from '../../components/FilterSelector/index.jsx';

const cx = classnames.bind(styles);

const FiltersMenu = ({ filter, setFilter, filters }) => (
  <>
    <ul className={cx('menu')}>
      { filters.map((option) => (
        <li
          key={uuidv4()}
          className={cx(
            'filter',
          )}
        >
          <FilterSelector
            setFilter={setFilter}
            filter={filter}
          >
            {option}
          </FilterSelector>
        </li>
      )) }
    </ul>
  </>
);

FiltersMenu.propTypes = {
  filter: PropTypes.string.isRequired,
  setFilter: PropTypes.func.isRequired,
  filters: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default FiltersMenu;
