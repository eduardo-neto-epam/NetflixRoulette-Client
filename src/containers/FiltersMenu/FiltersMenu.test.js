import React from 'react'
import '@testing-library/react'
import renderer from 'react-test-renderer';
import FiltersMenu from './index'

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

test('FiltersMenu renders correctly', () => {
    const tree = renderer
        .create(<FiltersMenu />)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });
