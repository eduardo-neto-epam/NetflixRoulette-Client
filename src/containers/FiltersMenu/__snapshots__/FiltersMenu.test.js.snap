// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersMenu renders correctly 1`] = `
<ul
  className="menu"
>
  <li
    className="filter"
  >
    <label
      className="filter"
      htmlFor="all"
    >
      all
      <input
        className="menu-filter"
        defaultChecked={true}
        id="all"
        name="filters"
        onChange={[Function]}
        type="radio"
        value="all"
      />
      <span
        className="checkmark"
      />
    </label>
  </li>
  <li
    className="filter"
  >
    <label
      className="filter"
      htmlFor="documentary"
    >
      documentary
      <input
        className="menu-filter"
        defaultChecked={false}
        id="documentary"
        name="filters"
        onChange={[Function]}
        type="radio"
        value="documentary"
      />
      <span
        className="checkmark"
      />
    </label>
  </li>
  <li
    className="filter"
  >
    <label
      className="filter"
      htmlFor="comedy"
    >
      comedy
      <input
        className="menu-filter"
        defaultChecked={false}
        id="comedy"
        name="filters"
        onChange={[Function]}
        type="radio"
        value="comedy"
      />
      <span
        className="checkmark"
      />
    </label>
  </li>
  <li
    className="filter"
  >
    <label
      className="filter"
      htmlFor="horror"
    >
      horror
      <input
        className="menu-filter"
        defaultChecked={false}
        id="horror"
        name="filters"
        onChange={[Function]}
        type="radio"
        value="horror"
      />
      <span
        className="checkmark"
      />
    </label>
  </li>
  <li
    className="filter"
  >
    <label
      className="filter"
      htmlFor="crime"
    >
      crime
      <input
        className="menu-filter"
        defaultChecked={false}
        id="crime"
        name="filters"
        onChange={[Function]}
        type="radio"
        value="crime"
      />
      <span
        className="checkmark"
      />
    </label>
  </li>
</ul>
`;
